pkg/util/winutil/eventlog/session/session.go

```diff
@@ -0,0 +1,93 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build windows
+
+// Package evtsession provides helpers for managing an Event Log API session
+// https://learn.microsoft.com/en-us/windows/win32/wes/accessing-remote-computers
+package evtsession
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/api"
+)
+
+// Session defines the interface for connecting to an Event Log host and is used to
+// manage event logs and query, subscribe, and render events.
+//
+// The Event Log API manages the connection behind the session handle. The connection to the
+// remote host is only made when the session is first used (e.g. EvtSubscribe).
+// If the remote host or the remote EventLog service is restarted the session handle
+// does not need to be closed and re-created.
+//
+// https://learn.microsoft.com/en-us/windows/win32/wes/accessing-remote-computers
+// https://learn.microsoft.com/en-us/windows/win32/api/winevt/nf-winevt-evtopensession
+type Session interface {
```

Comment:
For a local session, I think we could do that, but it's easier to just use `nil`.

For remote sessions we need more than one instance so it couldn't be a singleton/component. We could implement a factory/cache so different event log consumer can re-use sessions, but the Event Log API is implemented with RPC so that's essentially what already happens under the hood.
https://learn.microsoft.com/en-us/windows/win32/rpc/managing-network-connection-sets-associations-